swarm/pss/pss.go

```diff
@@ -644,7 +663,48 @@ func (self *Pss) send(to []byte, topic Topic, msg []byte, asymmetric bool, key [
 		Expire:  uint32(time.Now().Add(self.msgTTL).Unix()),
 		Payload: envelope,
 	}
-	return self.forward(pssmsg)
+	return self.enqueue(pssmsg)
+}
+
+func (self *Pss) enqueue(msg *PssMsg) error {
+	// check if message is expired
+	if self.isMsgExpired(msg) {
+		log.Trace("pss expired :/ ... dropping")
+		return errors.New("expired")
+	}
+	// flood guard:
+	// don't allow identical messages we saw shortly before
+	// cache the message
+	digest, err := self.storeMsg(msg)
+	if err != nil {
+		log.Warn(fmt.Sprintf("could not store message %v to cache: %v", msg, err))
+	}
+	if self.checkFwdCache(nil, digest) {
+		log.Trace(fmt.Sprintf("pss relay block-cache match: FROM %x TO %x", self.Overlay.BaseAddr(), common.ToHex(msg.To)))
```

Comment:
Nitpick, but it is easier to trace if the logs are structured. For example:

```
log.Trace("pss relay block-cache match", "from", self.Overlay.BaseAddr(), "to", common.ToHex(msg.To))
```

Reply:
Thanks I'll fix it in the next PR for deduplication (because this line doesn't exist there)